File: app/src/main/java/com/anthony/deepl/openl/util/AndroidUtils.java
package com.anthony.deepl.openl.util;

import android.content.ClipData;
import android.content.ClipDescription;
import android.content.ClipboardManager;
import android.content.Context;
import android.content.res.Resources;
import android.util.DisplayMetrics;
import android.util.TypedValue;

public class AndroidUtils {

    public static float convertDpToPixel(float dp, Context context) {
        Resources resources = context.getResources();
        return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, resources.getDisplayMetrics());
    }

    public static float convertPixelsToDp(float px, Context context) {
        Resources resources = context.getResources();
        DisplayMetrics metrics = resources.getDisplayMetrics();
        return px / (metrics.densityDpi / 160f);
    }

    public static String getClipboardText(ClipboardManager clipboardManager) {
        String text = null;
        if (clipboardManager != null && clipboardManager.hasPrimaryClip()) {
            ClipData data = clipboardManager.getPrimaryClip();
            if (data.getDescription().hasMimeType(ClipDescription.MIMETYPE_TEXT_PLAIN) || data.getDescription().hasMimeType(ClipDescription.MIMETYPE_TEXT_HTML)) {
                text = data.getItemAt(0).getText().toString();
            }
        }
        return (text != null && text.length() > 0) ? text : null;
    }

}

File: app/src/main/java/com/anthony/deepl/openl/util/AndroidUtils.kt
package com.anthony.deepl.openl.util

import android.content.ClipDescription
import android.content.ClipboardManager
import android.content.Context
import android.util.TypedValue

object AndroidUtils {

    fun convertDpToPixel(dp: Float, context: Context): Float {
        val resources = context.resources
        return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, resources.displayMetrics)
    }

    fun convertPixelsToDp(px: Float, context: Context): Float {
        val resources = context.resources
        val metrics = resources.displayMetrics
        return px / (metrics.densityDpi / 160f)
    }

    fun getClipboardText(clipboardManager: ClipboardManager?): String? {
        var text: String? = null
        if (clipboardManager != null && clipboardManager.hasPrimaryClip()) {
            val data = clipboardManager.primaryClip
            if (data.description.hasMimeType(ClipDescription.MIMETYPE_TEXT_PLAIN) || data.description.hasMimeType(ClipDescription.MIMETYPE_TEXT_HTML)) {
                text = data.getItemAt(0).text.toString()
            }
        }
        return if (!text.isNullOrEmpty()) text else null
    }

}

